linalg/sparsemat.cpp

```diff
@@ -1767,6 +1767,7 @@ void SparseMatrix::EliminateCol(int col, DiagonalPolicy dpolicy)
             if (aux->Column == col)
             {
                aux->Value = 0.0;
+               break;
```

Comment:
I'm not sure this is correct behavior. The linked list implementation of a sparse matrix allows for a column nonzero to be repeated in a given row, indicating the values should be summed. Thus by breaking early you miss zero-ing all the entries for that column.

Reply:
If it that so, it is definitely wrong. I believed that the columns are unique for each row as the add methods search the row first. I also do not see any summation in `Finalize()` if I am not wrong. Can you point me to a part doing that addition?

Reply:
Ah, you're probably right and I just assumed that was the case here. Apologies!

Reply:
No worries, but actually it is a good point. The/some methods could check internal consistency of the LIL if it does not contain repeated columns. At the moment, it goes under the radar if it happens somehow. A suggestion for a follow-up! 😉 Do you want to take it?